misc-scripts/xref_mapping/XrefParser/JGI_Parser.pm

```diff
@@ -76,89 +74,60 @@ sub run {
   #  DEDDDL*
   #
 
-
-
   my @xrefs;
 
-  local $/ = "\n>";
-
   my $file_io = $self->get_filehandle($file);
 
   if ( !defined $file_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
+    croak "ERROR: Could not open $file\n";
   }
 
-  while ( $_ = $file_io->getline() ) {
+  IO::Handle->input_record_separator( "\n>");
+  
+  while ( my $record = $file_io->getline() ) {
 
-    next if (/^File:/);   # skip header
+    next if ($record =~ /^File:/);   # skip header
 
     my $xref;
 
-    my ($header, $sequence) = $_ =~ /^>?(.+?)\n([^>]*)/s or warn("Can't parse FASTA entry: $_\n"); 
+    my ($header, $sequence) = $record =~ /^>?(.+?)\n([^>]*)/s or croak("Can't parse FASTA entry: $record\n");
+    my $accession = $header;
 
-      my @attr = split/\|/, $header ; 
-      my $acession = $header ;
-    
     # split header in different ways according to source name : 
-    my ( $version,$label  ) ;      
-    if ($source_name=~m/cint_jgi_v1/) { 
+
+    if ($source_name=~m/cint_jgi_v1/) {
       # header format is  >ci0100146277
-      # we want  146277
-      ($acession = $header)  =~s/\w{6}//;  
-      $version = "JGI 1.0" ;  
-    
-    } elsif ($source_name=~m/cint_aniseed_.*v1/) { 
+      # JGI 1.0
+      # we want accession 146277 from above
+      ($accession = $header) =~s/\w{6}//;
+
+    } elsif ($source_name=~m/cint_aniseed_.*v1/) {
```

Comment:
What is this  block ( elsif ($source_name=~m/cint_aniseed_.*v1/) { )  doing?!!

It's an empty block?!!  Do we need this?!

Reply:
Funnily enough, that's how this always was. I'll see what happens if I run it with the same regex as the previous block.